tests/Aspire.Hosting.Azure.Tests/AzureBicepResourceTests.cs

```diff
@@ -2781,6 +2781,121 @@ public async Task AddAzureOpenAI()
               properties: {
                 customSubDomainName: toLower(take(concat('openai', uniqueString(resourceGroup().id)), 24))
                 publicNetworkAccess: 'Enabled'
+                disableLocalAuth: true
+              }
+            }
+
+            resource roleAssignment_Hsk8rxWY8 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
+              scope: cognitiveServicesAccount_wXAGTFUId
+              name: guid(cognitiveServicesAccount_wXAGTFUId.id, principalId, subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'a001fd3d-188f-4b5d-821b-7da978bf7442'))
+              properties: {
+                roleDefinitionId: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'a001fd3d-188f-4b5d-821b-7da978bf7442')
+                principalId: principalId
+                principalType: principalType
+              }
+            }
+
+            resource cognitiveServicesAccountDeployment_5K9aRgiZP 'Microsoft.CognitiveServices/accounts/deployments@2023-05-01' = {
+              parent: cognitiveServicesAccount_wXAGTFUId
+              name: 'mymodel'
+              sku: {
+                name: 'Basic'
+                capacity: 4
+              }
+              properties: {
+                model: {
+                  format: 'OpenAI'
+                  name: 'gpt-35-turbo'
+                  version: '0613'
+                }
+              }
+            }
+
+            resource cognitiveServicesAccountDeployment_mdCAJJRlf 'Microsoft.CognitiveServices/accounts/deployments@2023-05-01' = {
+              parent: cognitiveServicesAccount_wXAGTFUId
+              dependsOn: [
+                cognitiveServicesAccountDeployment_5K9aRgiZP
+              ]
+              name: 'embedding-model'
+              sku: {
+                name: 'Basic'
+                capacity: 4
+              }
+              properties: {
+                model: {
+                  format: 'OpenAI'
+                  name: 'text-embedding-ada-002'
+                  version: '2'
+                }
+              }
+            }
+
+            output connectionString string = 'Endpoint=${cognitiveServicesAccount_wXAGTFUId.properties.endpoint}'
+            
+            """;
+        output.WriteLine(manifest.BicepText);
+        Assert.Equal(expectedBicep, manifest.BicepText);
+    }
+
+    [Fact]
```

Comment:
Would it make sense to turn the above test into a Theory instead? There appears to only be 2 lines different in these 2 tests.

Reply:
I'm happy to convert it to a Theory over a Fact. I tend to shy away from Theory if they result in conditional logic within the test itself, but that's more personal preference, so I'll defer to whatever is preferred by the aspire team.

Reply:
> if they result in conditional logic within the test itself

1 or 2 "if" checks vs. a hundred lines of duplication seems like a good tradeoff to me.

Reply:
I'll refactor the test and update the PR

Reply:
Test has been updated to use a theory.